### PR TITLE
Fix issues with tests not running deterministically

### PR DESCRIPTION
**Pull request by tiagojsag**
### Overview

_Please write a description. If the PR is hard to understand, provide a quick explanation of the code._

### Designs

_Link to the related design prototypes (if applicable)_

### Testing instructions

_Please explain how to test the PR: ID of a dataset, steps to reach the feature, etc._

### Feature relevant tickets

_Link to the related task manager tickets_

---

## Checklist before submitting

- [ ] Meaningful commits and code rebased on `develop`.
- [ ] If this PR adds feature that should be tested for regressions when
      deploying to staging/production, please add brief testing instructions
      to the deploy checklist (`docs/deployment-checklist.md`)
- [ ] Update CHANGELOG file

### CODE REVIEW

File: api/apps/geoprocessing/src/marxan-sandboxed-runner/adapters-single/solutions-output/solutions-output.service.ts [hunk 1]
```diff
@@ -21,6 +23,16 @@ export class SolutionsOutputService
     //
   }
 
+  interruptIfKilled = () => {
+    if (this.#cancelled) {
+      // TODO: do we need to cleanup anything?
```
Comment (tiagojsag) [inline api/apps/geoprocessing/src/marxan-sandboxed-runner/adapters-single/solutions-output/solutions-output.service.ts:28]: @hotzevzl I suspect we should review this and implement a proper cancel of this process... whatever it is that it does

File: api/apps/geoprocessing/src/marxan-sandboxed-runner/adapters-single/solutions-output/solutions-output.service.ts [hunk 2]
```diff
@@ -13,6 +13,8 @@ import { SandboxRunnerOutputHandler } from '../../ports/sandbox-runner-output-ha
 @Injectable()
 export class SolutionsOutputService
   implements SandboxRunnerOutputHandler<ExecutionResult> {
+  #cancelled = false;
```
Comment (kgajowy) [inline api/apps/geoprocessing/src/marxan-sandboxed-runner/adapters-single/solutions-output/solutions-output.service.ts:16]: AFAIR this service is injected once, so canceling one run would prevent all following runs to cancel the output dumping 🤔

File: api/apps/geoprocessing/src/marxan-sandboxed-runner/adapters-single/solutions-output/solutions-output.service.ts [hunk 3]
```diff
@@ -21,6 +23,16 @@ export class SolutionsOutputService
...
+      throw {
+        stdError: [],
+        signal: 'SIGTERM',
+      };
+    }
+  };
+
   /**
    * @Alicia notes:
    * for the output files we will want to first run a small script in python that will do some calcs we need to do for the 5 most different solutions; and later we will need to save in the db prior a transformation the next files: output_sum.csv, outputmv_xxxx.csv and outputr_xxxx.csv
```
Comment (hotzevzl) [inline api/apps/geoprocessing/src/marxan-sandboxed-runner/adapters-single/solutions-output/solutions-output.service.ts:38]: unrelated to the topic of this PR, but this whole comment can be deleted now as the workflow we ended up implementing is different and this comment is not out of date and not relevant anymore.

**Comment by tiagojsag** [comment]
rebased this with the blm branch, and added a few more fixes. I ran the tests 3-4 times on gh actions, and I am getting consistent greens - yay!

**Review by hotzevzl — APPROVED** [review]

**Pull request merged**